src/NServiceBus.Core/Transports/ReceiveBehavior.cs

```diff
@@ -41,5 +42,37 @@ public Registration(): base("ReceiveMessage", typeof(ReceiveBehavior), "Try rece
         }
     }
 
-   
+    /// <summary>
+    /// The raw message coming from the transport
+    /// </summary>
+    public class ReceivedMessage
+    {
+        /// <summary>
+        /// The native id of the message
+        /// </summary>
+        public string MessageId { get; private set; }
+
+        /// <summary>
+        /// The message headers
+        /// </summary>
+        public Dictionary<string, string> Headers { get; private set; }
+        
+        /// <summary>
+        /// The message body
+        /// </summary>
+        public byte[] Body { get; private set; }
```

Comment:
Makes sense. So we assume the transport handles us whatever stream instance they got from the underlying queue and we read it etc. What if queue's API provides byte[], not a stream?